harness/determined/profiler.py

```diff
@@ -532,3 +645,7 @@ def measure(self, batch_idx):
             except pynvml.NVMLError as e:
                 logging.info(f"{LOG_NAMESPACE}: failed to sample GPU memory for GPU {i}: {e}")
         return measurements
+
+
+def get_rank_if_horovod_process_else_return_zero() -> Optional[int]:
+    return int(os.getenv("HOROVOD_RANK", 0))
```

Comment:
Won't this mean that only one agent collects metrics?

Reply:
yeah true good catch, need local rank.

Reply:
or maybe it's a feature to not crash the master 🤣